### ISSUE TITLE
Pipeline info should include paths

### ISSUE THREAD

**Issue by ryanrichholt** [tgen/jetstream#78]
If running a pipeline, the pipeline info is available when rendering templates under the key:  `__pipeline__`. This comes from the loaded pipeline object manifest:

https://github.com/tgen/jetstream/blob/0a5099e1b637374cf3785f091d16e1f163e62676/jetstream/pipelines.py#L48

It would be better if the path to the pipeline was available, so that additional files included with the pipeline could be referenced easily. This was implemented to an extent with the bin directory. If a pipeline includes a bin directory, it will be added to $PATH when running the pipeline. But, this new idea opens up pipeline structures even more.  

Where to add it?

The `Pipeline.manifest` attribute is what gets included in the template rendering context currently. This is an exact copy of the loaded manifest file, and I think it's useful to leave unmodified. `Pipeline.info` was created as a quick way to access content in the manifest file `__pipeline__` field. I think the best solution is a new method for the Pipeline class, like `Pipeline.get_context()` that returns a copy of the manifest attribute with the paths added to the 
`__pipeline__` field. 

It could be used like this:

```
- cmd: cat "{{ __pipeline__.path }}/extras/large_data_file.txt

```

**Comment by ryanrichholt** [comment]
These variables should be exported to the environment so that they're resilient to changes with the location of the pipeline. Currently, the full path to the pipeline and files would be rendered to the cmd. If the pipeline was moved, all tasks would have a new identity when rendered and be forced to rerun.

**Comment by ryanrichholt** [comment]
Implemented in #93
